Branch-SDK/BNCSKAdNetwork.m

```diff
@@ -91,4 +101,94 @@ - (void)updatePostbackConversionValue:(NSInteger)conversionValue
     
 }
 
+- (void)updatePostbackConversionValue:(NSInteger)fineValue
+                          coarseValue:(NSString *)coarseValue
+                           lockWindow:(BOOL)lockWindow
+                    completionHandler:(void (^)(NSError *error))completion {
+    if (@available(iOS 16.1, *)) {
+        if ([self shouldAttemptSKAdNetworkCallout]) {
+            
+            ((id (*)(id, SEL, NSInteger, NSString *, BOOL, void (^)(NSError *error)))[self.skAdNetworkClass methodForSelector:self.skAdNetworkUpdatePostbackConversionValueCoarseValueLockWindow])(self.skAdNetworkClass, self.skAdNetworkUpdatePostbackConversionValueCoarseValueLockWindow, fineValue, coarseValue, lockWindow, completion);
+        }
+    }
+}
+
+- (int) calculateSKANWindowForTime:(NSDate *) currentTime{
+    
+    int firstWindowDuration = 2 * 24 * 3600;
+    int secondWindowDuration = 7 * 24 * 3600;
+    int thirdWindowDuration = 35 * 24 * 3600;
+    
+    NSTimeInterval timeDiff = [currentTime timeIntervalSinceDate:self.installDate];
+    
+    if (timeDiff <= firstWindowDuration) {
+        return 1;
+    } else if (timeDiff <= secondWindowDuration) {
+        return 2;
+    }else if (timeDiff <= thirdWindowDuration) {
+        return 3;
+    }
+    return 0;
+}
+
+- (NSString *) getCoarseConversionValueFromDataResponse:(NSDictionary *) dataResponseDictionary{
+    
+    NSString *coarseConversionValue = dataResponseDictionary[BRANCH_RESPONSE_KEY_COARSE_KEY] ;
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 160100
```

Comment:
why are we using this syntax over     if (@available(iOS 16.1, *)) unless this is just how github is reporting it.

Reply:
@jf-branch Imported it for enum values SKAdNetworkCoarseConversionValueHigh etc. We can replace them with string constants. But though Apple can change their values ... so its better to use enum constants. For function call we are using function pointers. We couldn't fetch pointers to string Constants. Correct me if I am wrong or there is way to do so.

Reply:
After our discussion yesterday, I feel just using the raw string values will be ok.

Reply:
Yes ! In last commit, I did remove Storekit framework and using raw strings.